### DISCUSSION TITLE
Testing assembly - problem with random resets

### DISCUSSION THREAD

**Discussion by jgalinat** [thehighnibble/thehighnibble.github.io#60] [IMSAI8080esp replica]
Hi recently purchased a High Nibble IMSAI8080esp kit from another enthusiast off of eBay.  I have assembled the kit just short of soldering in the toggle switches. Everything seems to work OK when I do front panel operations such as reset & examine next using the jumper wire and the web interface gets served from the SD card.   I am having a problem with what seems to be spurious resets of the ESP32 pico board.  The front panel board goes blank for a few seconds and the web interface says “reloading”. When it comes back it works fine until the next random reset. I assumed it was power related but it occurs with 3 different power supplies including a 2.4amp model I use with RaspPis.  I reflowed the solder on the female connectors for the ESP32 board and the 6 pins I added to the ESP32 board with no change.  Have any of you folks seen this before?  In case it is the ESP32Pico board, I have bought a spare but obviously I would have to load the firmware on a new board and I am unsure the best way to accomplish that. Any help or advice would be appreciated. 
Thanks,
Jeff Galinat

**Comment by GrantMeStrength** [comment]
I found that an incomplete kit can act in unexpected ways, but once it's all soldered together it's a lot more stable. I would continue soldering the switches if I were you.

If you want to flash a different ESP, the [guide](https://thehighnibble.com/imsai8080/update/#overview) does a fair job at explaining the process. You'll need to install Python tools and set aside some time to get it working. Remember to remove the ESP from the IMSAI PCB before trying it. 

If there are issues with the construction they're likely to be around the SMD components (the ram and the SD card reader) and a re-flow can do wonders (and checking everything with a multimeter).

**Comment by jgalinat** [comment]
Hi John
  Thanks for the reply!  I will proceed and see how things work out.  I
have ohmed out my solder joints multiple times and inspected them with a
jeweler’s loupe and they look good. I will reflow the surface mount pins if
I still have problems after assembly.  At least I can get to them from the
back of the board.  It fails just like the reset button on the rear of the
board is pressed then the code reloads and all is well until the next
spurious reset.   I have read that guide from the site you linked to but it
discusses updating the firmware on a operating system not starting from a
blank slate. I have very little experience with most of these
microcontroller based boards. I have used Raspberry Pi’s but only as a
Linux replacement for a small PC. I will keep digging.
Thanks for your suggestions
Jeff

**Reply by GrantMeStrength** [reply]
The page does cover using python tools to write the firmware to the ESP, but indirectly - the guide in is the file called FLASHING.MD in the associated firmware download from GitHub repo. But you probably don't need it - just keep soldering!

**Reply by jgalinat** [reply]
Hi John,
  I muddled through python and the esptools/espefuse scripts and finally got the job done loading the latest code onto the spare ESP32Pico kit I got off Amazon. I played with the board for at least 1/2 hour with no spurious resets so I think the problem is fixed. I think I can now proceed with the rest of the build without lingering doubts. Thanks again for pointing me to the correct software tools. 
 Regards,
Jeff

**Comment by jgalinat** [comment]
Hi John,
  I must have missed it. I will reread it. Thank you!
Jeff

**Comment by jgalinat** [comment]
Hi John,
  Found it! Thanks!
